### PR TITLE
 BaSP-MR-198: Logic and validations for class subscribed and capacity

### PR DESCRIPTION
**Pull request by jramire5**
Now, it should add one to subscribed on class when a member subscribes to a class. And rest one when a member unsubscribes. If a class has equal subscribed members to capacity, it shouldn't let subscribe to this class

Please check this PR using the MR-198 branch in the server

### CODE REVIEW

**Review by josedaniele — CHANGES_REQUESTED** [review]
When I subscribe to a class and then unsubscribe, the number of subscribed members does not decrease, it remains the same.
step 1 i subscribe to football
image

step 2 the numbers of subscribed increase in 1
image
step 3 i unsubscribed of the class
image
step 4 the class has the same number of enrollees
image

**Comment by jramire5** [comment]
@josedaniele I think this is solved

**Review by josedaniele — APPROVED** [review]

File: src/Components/Admins/Subscriptions/Form/index.jsx [hunk 1]
```diff
@@ -67,13 +68,23 @@ const Form = () => {
...
+    console.log(data);
     try {
       if (id) {
         editSubscription(dispatch, data, id);
         setShowConfirmModal(false);
       } else {
-        addSubscriptions(dispatch, data);
-        setShowConfirmModal(false);
+        const subClass = classes.find((c) => c._id === data.classes) || '';
+        console.log(subClass);
```
Comment (josedaniele) [inline src/Components/Admins/Subscriptions/Form/index.jsx]: delete this console.log

File: src/Components/Admins/Subscriptions/Table/index.jsx [hunk 2]
```diff
@@ -10,21 +10,28 @@ import ConfirmModal from 'Components/Shared/ConfirmModal';
...
   const handleConfirmDelete = (subscriptionId) => {
     setEditingSubscriptionId(subscriptionId);
     setShowConfirmDeleteModal(true);
   };
 
-  const handleDelete = (subscriptionId) => {
-    dispatch(deleteSubscription(subscriptionId));
+  const handleDelete = (subscription) => {
+    const subClass = classes.find((subClass) => subClass._id === subscription.classes._id);
+    console.log(subClass);
```
Comment (josedaniele) [inline src/Components/Admins/Subscriptions/Table/index.jsx]: and this

File: src/Components/Members/SubscriptionsMember/index.jsx [hunk 3]
```diff
@@ -54,7 +56,12 @@ const SubscriptionsMember = () => {
   };
 
   const handleDeleteActivity = () => {
+    const subscription = subscriptions.find((sub) => sub._id === idToDelete) || '';
+    const unsubscribed = subscription.classes.subscribed - 1;
+    const subscriptionEdited = { subscribed: unsubscribed };
     dispatch(deleteSubscription(idToDelete));
+    dispatch(addSubscribed(subscriptionEdited, subscription.classes._id));
+    console.log(subscriptionEdited);
```
Comment (josedaniele) [inline src/Components/Members/SubscriptionsMember/index.jsx]: and this

File: src/Redux/Classes/thunks.js [hunk 4]
```diff
@@ -100,6 +100,35 @@ export const putClass = async (dispatch, classes, id, history) => {
...
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/class/${id}`, {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json',
+          token: token
+        },
+        body: JSON.stringify(classes)
+      });
+      const { data, message, error } = await response.json();
+      console.log('response', response);
```
Comment (josedaniele) [inline src/Redux/Classes/thunks.js]: and this

**Review by josedaniele — CHANGES_REQUESTED** [review]

**Review by IgnacioGalceran — APPROVED** [review]
nice!

**Review by MauroCaffesse — APPROVED** [review]

**Review by josedaniele — APPROVED** [review]

**Pull request merged**